### ISSUE TITLE
Issue with sorting

### ISSUE THREAD

**Issue by han9over** [delannoyk/AudioPlayer#40]
@delannoyk 

I think there is an issue with the way songs are sorted. I think you will be able to replicate this if you follow below steps.

1. Add around 2-3 songs separately by calling addItemToQueue
2. Use tableview to show your list of songs. When a song is clicked on tableview, use below code to play that song

```
if let loadedItems = player.items {
                player.playItems(loadedItems, startAtIndex: indexPath.row)
}
```

3. Now, add another couple songs to the queue using same addItemToQueue function.
4. Now try to click on tableview and you will see that the songs are out of sequence.

I think the problem is with sort function. When the song is added to queue, the position will be 0. When the song is played using playItems, the positions are applied. Next when you add items to the queue again, they will add with 0 and the new items go to the top of list.

Am I clear?

**Comment by delannoyk** [comment]
Hi @han9over 
Yep I see the issue. If I understand correctly, new items are being added to the queue with `position` equals to 0, so multiple items are positioned at index 0.
New coming version will not suffer from this issue. Anyway, it doesn't have an ETA yet and will break Swift 2.0 compatibility, so I'll try to work on a quick fix ASAP for this.

**Comment by han9over** [comment]
Thanks. That's totally fine. I already have workaround in my project for this. So I'm good. Just wanted to report it out to you. :)

**Comment by han9over** [comment]
My work around is to remove sorting as I don't see a need for it as I don't need shuffle.

**Comment by delannoyk** [comment]
Fixed in 0.8.2 :)
